crates/db_views/src/comment/comment_view.rs

```diff
@@ -256,6 +271,7 @@ impl CommentQuery<'_> {
       // + !post_id isn't used anyways (afaik)
       if o.post_id.is_some() || o.parent_path.is_some() {
         // Always order by the parent path first
+        // TODO not sure if this initial then_order_by will work correctly
```

Comment:
Doing any sorting before creating the paginated query builder is always incorrect.

Reply:
This one I couldn't figure out. Problem is [here](https://github.com/LemmyNet/lemmy/pull/5429/files#diff-02f588aeca2afd55626f0b644b6741bd3d16359b0b26d3853af829411537e312R314), trying do a subpath sort:

`pq = pq.then_order_by(subpath(key::path, 0, -1));`

Reply:
That can be done using a new struct similar to ReverseTimestampKey but it would call subpath(x, 0, -1) instead of reverse_timestamp_key(x). If it needs to be in a different direction than the direction that PaginatedQueryBuilder is initialized with, then something more complicated is needed, which I could try to figure out when I have time.

Reply:
If I understand correctly, it doesn't matter if the subpath sort is ascending or descending, just like the id tie-breaker at the end

Reply:
Also, to prevent the confusion that I had, add a comment saying that putting comments underneath their parents is done by the frontend instead of in the db query